### PR TITLE
Make Avalonia.Browser work on .NET 8

### LINKED ISSUE #12225: Avalonia Browser is broken on net8.0 (Emscripten 3.1.34)
**Describe the bug**
Wasm app is broken when use latest dotnet sdk 8.0 which use Emscripten 3.1.34

**To Reproduce**
1. Install dotnet sdk 8.0.100-preview.6.23330.14
2. Upgrade your wasm projet to net8.0
3. Build.
4. See Error in MSBuild.log
1>wasm-ld : warning : function signature mismatch: time

**Expected behavior**
It works

**Logs**
wasm-ld : warning : function signature mismatch: time
error MSB3073: The command "emcc exited with code 1.

**Desktop:**

- OS: Microsoft Windows 11 (22621)
- SDK : dotnet 8.0.100-preview.6.23330.14
- Avalonia : 11.0

**Additional context**
 
https://github.com/mono/SkiaSharp/issues/2494
https://github.com/mono/SkiaSharp/pull/2495

### LINKED ISSUE #13219: Avalonia iOS is broken on net8.0-rc2
## Describe the bug
iOS app is broken when use latest dotnet sdk 8.0-rc2.

## To Reproduce

1. Install dotnet sdk 8.0.100-rc.2.23502.2
2. Create a new project with `dotnet new avalonia.xplat -o GetStartedApp`
3. Upgrade all projects to net8.0
4. `sudo dotnet worload restore`
5. `dotnet build`

## Expected behavior
It works

## Logs
```
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : clang++ exited with code 1: [/Users/baochenw/Src/Avalonia/GetStartedApp/GetStartedApp.iOS/GetStartedApp.iOS.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : ld: warning: ignoring duplicate libraries: '-lSystem.Globalization.Native', '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmono-component-marshal-ilgen', '-lmonosgen-2.0', '-lxamarin-dotnet-debug' [/Users/baochenw/Src/Avalonia/GetStartedApp/GetStartedApp.iOS/GetStartedApp.iOS.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : ld: warning: mixed ObjC ABI, /usr/local/share/dotnet/packs/Microsoft.iOS.Runtime.iossimulator-arm64/16.4.8968-net8-rc2/runtimes/iossimulator-arm64/native/Microsoft.iOS.registrar.a compiled with category class properties [/Users/baochenw/Src/Avalonia/GetStartedApp/GetStartedApp.iOS/GetStartedApp.iOS.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : ld: warning: mixed ObjC ABI, /Users/baochenw/Src/Avalonia/GetStartedApp/GetStartedApp.iOS/obj/Debug/net8.0-ios/iossimulator-arm64/nativelibraries/main.arm64.o compiled with category class properties [/Users/baochenw/Src/Avalonia/GetStartedApp/GetStartedApp.iOS/GetStartedApp.iOS.csproj]
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.8968-net8-rc2/targets/Xamarin.Shared.Sdk.targets(1556,3): error : ld: building for 'iOS-simulator', but linking in dylib (/Users/baochenw/.nuget/packages/harfbuzzsharp.nativeassets.ios/2.8.2.3/runtimes/ios/native/libHarfBuzzSharp.framework/libHarfBuzzSharp) built for ' [/Users/baochenw/Src/Avalonia/GetStartedApp/GetStartedApp.iOS/GetStartedApp.iOS.csproj]

```

## Environment 
```
dotnet --info
.NET SDK:
 Version:   8.0.100-rc.2.23502.2
 Commit:    0abacfc2b6

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.6
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.100-rc.2.23502.2/

.NET workloads installed:
 [wasm-tools]
   Installation Source: SDK 8.0.100-rc.2
   Manifest Version:    8.0.0-rc.2.23479.6/8.0.100-rc.2
   Manifest Path:       /usr/local/share/dotnet/sdk-manifests/8.0.100-rc.2/microsoft.net.workload.mono.toolchain.current/8.0.0-rc.2.23479.6/WorkloadManifest.json
   Install Type:        FileBased

 [macos]
   Installation Source: SDK 8.0.100-rc.2
   Manifest Version:    13.3.8968-net8-rc2/8.0.100-rc.2
   Manifest Path:       /usr/local/share/dotnet/sdk-manifests/8.0.100-rc.2/microsoft.net.sdk.macos/13.3.8968-net8-rc2/WorkloadManifest.json
   Install Type:        FileBased

 [ios]
   Installation Source: SDK 8.0.100-rc.2
   Manifest Version:    16.4.8968-net8-rc2/8.0.100-rc.2
   Manifest Path:       /usr/local/share/dotnet/sdk-manifests/8.0.100-rc.2/microsoft.net.sdk.ios/16.4.8968-net8-rc2/WorkloadManifest.json
   Install Type:        FileBased

 [android]
   Installation Source: SDK 8.0.100-rc.2
   Manifest Version:    34.0.0-rc.2.468/8.0.100-rc.2
   Manifest Path:       /usr/local/share/dotnet/sdk-manifests/8.0.100-rc.2/microsoft.net.sdk.android/34.0.0-rc.2.468/WorkloadManifest.json
   Install Type:        FileBased


Host:
  Version:      8.0.0-rc.2.23479.6
  Architecture: arm64
  Commit:       0b25e38ad3

.NET SDKs installed:
  6.0.404 [/usr/local/share/dotnet/sdk]
  6.0.408 [/usr/local/share/dotnet/sdk]
  6.0.412 [/usr/local/share/dotnet/sdk]
  6.0.414 [/usr/local/share/dotnet/sdk]
  7.0.101 [/usr/local/share/dotnet/sdk]
  7.0.203 [/usr/local/share/dotnet/sdk]
  7.0.302 [/usr/local/share/dotnet/sdk]
  7.0.306 [/usr/local/share/dotnet/sdk]
  7.0.308 [/usr/local/share/dotnet/sdk]
  8.0.100-rc.1.23463.5 [/usr/local/share/dotnet/sdk]
  8.0.100-rc.2.23502.2 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.12 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.20 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.22 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-rc.1.23421.29 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-rc.2.23480.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.12 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.9 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-rc.1.23419.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-rc.2.23479.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

- OS: Mac M1 13.6
- Avalonia-Version: 11.0.4

## Additional context
[GetStartedApp.tar.gz](https://github.com/AvaloniaUI/Avalonia/files/12878525/GetStartedApp.tar.gz)

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

There were multiple problems with Avalonia on .NET WASM 8:
1. Avalonia used wrong Emscripten version (easy to fix).
2. Avalonia used `new Timer` which for some reason doesn't work in .NET WASM 8 anymore (todo: need to investigate). "setTimeout" was used instead as it's a better option for us anyway.
3. Task.ContinueWith with default arguments for some reason doesn't work in .NET WASM 8 anymore (todo: need to investigate). Adding explicit arguments everywhere makes it work.

Other changes, not much related to .NET 8:
1. UseAvaloniaWasmDefaultOptimizations was removed. Our custom wasm parameters were conflicting with .NET defaults, like support for globalization (which now works way better than before in .NET 6).
2. ControlCatalog.Browser project was updated from our default templates.

NOTE: This PR DOES NOT add multithreading support on WASM Avalonia. It might work with user code, but Avalonia itself still works in single-threaded mode on WASM.
This PR also doesn't change repository SDK or samples TFM, so it still works with .NET 7.

## Breaking changes

.NET 8 WASM has some breaking changes by itself, like "./dotnet.js" file was moved to "./_framework/dotnet.js". And same with "avalonia.js" now as well.

## Fixed issues

Fixes #12225
Fixes #13219 (by updating HarfBuzzSharp)

### CODE REVIEW

File: src/Browser/Avalonia.Browser/WindowingPlatform.cs [hunk 1]
```diff
@@ -70,23 +67,22 @@ public void Signal(DispatcherPriority priority)
...
+            var interval = 1;
+
+            CanvasHelper.SetTimeout(() =>
             {
                 _signaled = false;
-                disp?.Dispose();
-
                 Signaled?.Invoke(null);
-            }, null, interval, interval);
+            }, interval);
```
Comment (maxkatz6) [inline src/Browser/Avalonia.Browser/WindowingPlatform.cs:78]: setTimeout runs only once, so there is no need to dispose itself.

**Comment by hez2010** [comment]
> Avalonia used new Timer which for some reason doesn't work in .NET WASM 8 anymore

Tested in .NET 8 rc2 and seems that Timer works as expected?

image

```cs
@code {
    protected override void OnInitialized()
    {
        var timer = new Timer((obj) => Console.WriteLine("tick"), null, 0, 1000);
    }
}
```

**Review by Gillibald — APPROVED** [review]

**Comment by Mrxx99** [comment]
Will this be merged and released in time for .NET 8 release?

**Comment by maxkatz6** [comment]
> Will this be merged and released in time for .NET 8 release?

I will try to push backport candidate on it, but otherwise, it's planned to be in 11.1, which is going to be after .NET 8 release.

**Comment by maxkatz6** [comment]
> Tested in .NET 8 rc2 and seems that Timer works as expected?

Yes, our guess was that timer now callbacks on sync context, which is blocked by Dispatcher awaiting for the next timer tick... So pretty much avalonia-specific problem. Using lower level platform setTimeout/interval is more expected behavior for the dispatcher either way.

**Comment by curia-damiano** [comment]
Thank you for the work on this PR. In which version of the NuGet can we expect this fix?

**Comment by timunie** [comment]
> In which version of the NuGet can we expect this fix?

Hopefully 11.0.6, latest 11.1

Nightly builds have this fix since it was merged, so you are welcome to test nightly until released into stable.

**Comment by curia-damiano** [comment]
Thank you very much for the info!
I will patiently wait for the NuGet packages :-)

**Comment by timunie** [comment]
Nightly has nuget packages, it is just a different feed url. https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed

**Comment by curia-damiano** [comment]
Thank you for the tip!
I have added the package source, upgraded my NuGets to 11.0.999, and I get errors: undefined symbol std:__2:__vector_base_common<true>::__throw_length_error() const

But maybe it's something on my setup, I can wait for the public version, thank you.

**Comment by MrJul** [comment]
Please check the *pre-release* checkbox and use the latest, which should be 11.**1**.999-cibuild00xxxxx-beta.

**Comment by HendrikMennen** [comment]
On MacOS, probably on Linux too, the paths for avalonia.js and some others look broken:
image
Probably hardcoded the windows directory separator, which is why it fails on unix

**Pull request merged**